cfme/tests/cloud/test_quota_tagging.py

```diff
@@ -0,0 +1,181 @@
+        instances.instantiate('quota_source')
+    original_instance.copy_to(domain=test_domain)
+
+    original_instance = miq. \
+        namespaces.instantiate('System'). \
+        namespaces.instantiate('CommonMethods'). \
+        classes.instantiate('QuotaStateMachine'). \
+        instances.instantiate('quota')
+    original_instance.copy_to(domain=test_domain)
+
+    instance = test_domain. \
+        namespaces.instantiate('System'). \
+        namespaces.instantiate('CommonMethods'). \
+        classes.instantiate('QuotaStateMachine'). \
+        instances.instantiate('quota')
+    return instance
+
+
+def set_entity_quota_source(max_quota_test_instance, entity):
+    with update(max_quota_test_instance):
+        max_quota_test_instance.fields = {'quota_source_type': {'value': entity}}
+
+
+@pytest.fixture(params=[('groups', 'group', 'EvmGroup-super_administrator'),
+                        ('users', 'user', 'Administrator')], ids=['group', 'user'], scope='module')
+def entities(appliance, request, max_quota_test_instance):
+    collection, entity, description = request.param
+    set_entity_quota_source(max_quota_test_instance, entity)
+    return getattr(appliance.collections, collection).instantiate(description)
+
+
+@pytest.fixture(scope='function')
+def set_entity_quota_tag(request, entities, appliance):
+    tag, value = request.param
+    tag = Tag(display_name=value, category=Category(display_name=tag))
+    entities.add_tag(tag)
+    yield
+    # will refresh page as navigation to configuration is blocked if alert are on requests page
+    appliance.server.browser.refresh()
+    entities.remove_tag(tag)
+
+
+@pytest.mark.parametrize(
+    ['set_entity_quota_tag'],
+    [
+        [('Quota - Max Memory *', '1GB')],
+        [('Quota - Max Storage *', '10GB')],
+        [('Quota - Max CPUs *', '1')]
+    ],
+    indirect=['set_entity_quota_tag'],
+    ids=['max_memory', 'max_storage', 'max_cpu']
+)
+def test_quota_tagging_cloud_via_lifecycle(request, appliance, provider, prov_data, setup_provider,
+                                           set_entity_quota_tag, template_name, vm_name):
+    """Test Group and User Quota in UI using tagging"""
+    recursive_update(prov_data, {
+        'request': {'email': 'test_{}@example.com'.format(fauxfactory.gen_alphanumeric())}})
+    prov_data.update({'template_name': template_name})
+    appliance.collections.cloud_instances.create(vm_name, provider, prov_data)
+    # nav to requests page to check quota validation
+    request_description = 'Provision from [{}] to [{}]'.format(template_name, vm_name)
+    provision_request = appliance.collections.requests.instantiate(request_description)
+    provision_request.wait_for_request(method='ui')
+    request.addfinalizer(provision_request.remove_request)
+    assert provision_request.row.reason.text == "Quota Exceeded"
+
+
+@pytest.mark.parametrize('context', [ViaSSUI, ViaUI])
+@pytest.mark.parametrize(
+    ['set_entity_quota_tag'],
+    [
```

Comment:
**optional** use `param=...` on the fixture declaration as the parameterization as is is duplicate